File: backend/models/withdrawal.go
package models
import (
	"time"
	"github.com/google/uuid"
	"gorm.io/gorm"
)
type Withdrawal struct {
	ID            uuid.UUID      `gorm:"type:uuid;primary_key" json:"id"`
	UserID        uuid.UUID      `gorm:"type:uuid;not null;index" json:"user_id"`
	Amount        float64        `gorm:"type:decimal(15,2);not null" json:"amount" validate:"required,gt=0"`
	BankName      string         `gorm:"type:varchar(100);not null" json:"bank_name" validate:"required"`
	AccountNumber string         `gorm:"type:varchar(50);not null" json:"account_number" validate:"required"`
	AccountName   string         `gorm:"type:varchar(255);not null" json:"account_name" validate:"required"`
	Status        string         `gorm:"type:varchar(50);default:'pending'" json:"status"`
	Notes         string         `gorm:"type:text" json:"notes"`
	ProcessedAt   *time.Time     `json:"processed_at"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
	User User `gorm:"foreignKey:UserID" json:"user,omitempty"`
}
func (w *Withdrawal) BeforeCreate(tx *gorm.DB) error {
	if w.ID == uuid.Nil {
		w.ID = uuid.New()
	}
	return nil
}

File: fluttermultiplatform/lib/models/withdrawal.dart
import 'user.dart';

class Withdrawal {
  final String id;
  final String userId;
  final double amount;
  final String bankName;
  final String accountNumber;
  final String accountName;
  final String status;
  final String notes;
  final String? processedAt;
  final User? user;
  final String createdAt;

  Withdrawal({required this.id, this.userId = '', this.amount = 0, this.bankName = '', this.accountNumber = '', this.accountName = '', this.status = '', this.notes = '', this.processedAt, this.user, this.createdAt = ''});

  factory Withdrawal.fromJson(Map<String, dynamic> json) => Withdrawal(
        id: json['id']?.toString() ?? '',
        userId: json['user_id']?.toString() ?? '',
        amount: (json['amount'] ?? 0).toDouble(),
        bankName: json['bank_name'] ?? '',
        accountNumber: json['account_number'] ?? '',
        accountName: json['account_name'] ?? '',
        status: json['status'] ?? '',
        notes: json['notes'] ?? '',
        processedAt: json['processed_at'],
        user: json['user'] != null ? User.fromJson(json['user']) : null,
        createdAt: json['created_at'] ?? '',
      );
}
